### DISCUSSION TITLE
Add code coverage checks

### DISCUSSION THREAD

**Discussion by GeorgeLeithead** [CommunityToolkit/Maui#555] [General]
I know that Vlad is/was working on adding commit hooks to ensure code formatting `dotnet format`, see [How to automatically keep your code clean](https://vladislavantonyuk.azurewebsites.net/articles/How-to-automatically-keep-your-code-clean)  Could we also possibly add Code Coverage checks? Such as [ASP.NET Core - Implementing Code Coverage with JetBrains dotCover](https://digitaldrummerj.me/aspnet-core-code-coverage/)

**Comment by GeorgeLeithead** [comment]
There is, I just didn't realise!

In the PR, in the cheks, simply click on the checks, then in the report details click on "Code coverage", then in the Azure DevOps page, click on the "Code coverage" tab!

**Comment by VladislavAntonyuk** [comment]
Yes, we already have it. We use coverlet to calculate code coverage

**Reply by GeorgeLeithead** [reply]
I clearly didn't look hard enough.  Perhaps this could be added to the "Contributing" to get everyone to look to make sure that they are aware, and that they should check to make sure their tests are good?

**Reply by VladislavAntonyuk** [reply]
It should be available on the Actions page

**Comment by VladislavAntonyuk** [comment]
Brandon added code format warnings to the pipeline. Enabling git hooks for automatic code formatting requires configuration and GitHooks package dependency as I described in the article. For now we do not plan to add additional package

**Reply by GeorgeLeithead** [reply]
I read the article.  Looks good.  From what I can (now that I know where it is) see, the coverage reports are prety good, and I'm already using it to improve coverage.

**Comment by GeorgeLeithead** [comment]
Just read this article.
[Improving MAUI & Xamarin Code Quality with Pre Git Commit Linting](https://prototypemakers.medium.com/improving-maui-xamarin-code-quality-with-pre-git-commit-linting-e3e24ce585d9). Could we add this Pre Git Commit?

**Reply by pictos** [reply]
From the article, we need to install node locally in order to use this. I don't want to install node on my machine just because of that... And maybe other contributors don't want as well. I believe that `editorconfig` will be enough

**Reply by VladislavAntonyuk** [reply]
It is actually the same as using GitHooks nuget package just to configure git hooks.
For now we do not plan to add it to the library. But you can definitely setup it on your machine to ensure you always publish formatted code

**Comment by CliffAgius** [comment]
Closed as answered
